src/3d/qgs3dmapsettings.h

```diff
@@ -446,6 +459,7 @@ class _3D_EXPORT Qgs3DMapSettings : public QObject
     QgsPathResolver mPathResolver;
     QgsMapThemeCollection *mMapThemes = nullptr;   //!< Pointer to map themes (e.g. from the current project) to resolve map theme content from the name
     double mDpi = 96;  //!< Dot per inch value for the screen / painter
+    QgsMesh3DSymbol mMeshterrainSymbol;
```

Comment:
Why is mesh terrain symbol defined here? I think it should be defined inside the mesh terrain generator...

Reply:
If this symbol is defined in the mesh terrain generator, it would be destruct if the terrain generator is changed. Here, it could be saved event if the terrain is changed.

Reply:
I think it is fine if the symbol gets "lost" when changing to a different terrain generator. If the symbol would be inside the mesh terrain generator, it would be cleaner solution.